nncf/torch/model_creation.py

```diff
@@ -117,19 +132,18 @@ def create_compressed_model(model: Module, config: NNCFConfig,
                                    target_scopes=target_scopes,
                                    scopes_without_shape_matching=scopes_without_shape_matching)
 
-    should_init = resuming_state_dict is None
-    composite_builder = PTCompositeCompressionAlgorithmBuilder(config, should_init=should_init)
+    should_init = compression_state is None
+    composite_builder = CompositeBuilder(config, should_init, compression_setups)
     composite_builder.apply_to(compressed_model)
-
     compression_ctrl = composite_builder.build_controller(compressed_model)
 
     # Required to ensure that the model leaving create_compressed_model has correct compressed graph.
     # In particular, this is currently required for correct functioning of RNNs.
     compressed_model.rebuild_graph()
 
     try:
-        if resuming_state_dict is not None:
-            load_state(compressed_model, resuming_state_dict, is_resume=True)
+        if model_state is not None:
+            load_state(compressed_model, model_state, is_resume=is_strict)
```

Comment:
As agreed, almost in all basic scenarios load in a strict mode. if user need to add or remove algorithm, `is_strict` should explicitly set to `False`.